bookstore/publish.py

```diff
@@ -64,19 +90,30 @@ def initialize(self):
             region_name=self.bookstore_settings.s3_region_name,
         ) as client:
             self.log.info("Processing published write of %s", path)
-            obj = await client.put_object(
-                Bucket=self.bookstore_settings.s3_bucket,
-                Key=s3_object_key,
-                Body=json.dumps(model['content']),
-            )
+            try:
+                obj = await client.put_object(
+                    Bucket=self.bookstore_settings.s3_bucket,
+                    Key=s3_object_key,
+                    Body=json.dumps(content),
+                )
+            except ClientError as e:
+                status_code = e.response['ResponseMetadata'].get('HTTPStatusCode')
+                raise web.HTTPError(status_code, e.args[0])
             self.log.info("Done with published write of %s", path)
 
-        self.set_status(201)
+        resp_content = self.prepare_response(path, obj)
+        resp_str = json.dumps(resp_content)
```

Comment:
Perhaps:

`return json.dumps(resp_content)`

Reply:
that's what I'd originally had but was probably trying to overoptimize for new contributor legibility.

Reply:
Fair enough. I typically lean toward less lines is less code to maintain after understanding. In this particular case, either is fine. If the lines were several away from each other, I would lean toward being explicit to avoid confusion. Since they are close together, I tend to opt to leave out the placeholder variable.